From 96dd9524fa70a92f7dee564de5cc182502bd870c Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:43:40 -0400
Subject: [infiniband] cxgb4: Initialize the device status page

Message-id: <1410475447-94392-82-git-send-email-svemuri@redhat.com>
Patchwork-id: 93816
O-Subject: [RHEL7.1 PATCH BZ 1124947 081/108] RDMA/cxgb4: Initialize the device status page
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

The status page is mapped to user processes and allows sharing the
device state between the kernel and user processes.  This state isn't
getting initialized and thus intermittently causes problems.  Namely,
the user process can mistakenly think the user doorbell writes are
disabled which causes SQ work requests to never get fetched by HW.

Fixes: 05eb23893c2c ("cxgb4/iw_cxgb4: Doorbell Drop Avoidance Bug Fixes").
Signed-off-by: Steve Wise <swise@opengridcomputing.com>
Cc: <stable@vger.kernel.org> # v3.15
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 6b54d54dea82ae214e4a45a503c4ef755a8ecee8)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/cxgb4/device.c b/drivers/infiniband/hw/cxgb4/device.c
index dd93aad..16b75de 100644
--- a/drivers/infiniband/hw/cxgb4/device.c
+++ b/drivers/infiniband/hw/cxgb4/device.c
@@ -696,6 +696,7 @@ static int c4iw_rdev_open(struct c4iw_rdev *rdev)
   pr_err(MOD "error allocating status page\n");
   goto err4;
  }
+ rdev->status_page->db_off = 0;
  return 0;
 err4:
  c4iw_rqtpool_destroy(rdev);
-- 
1.7.1